include/seastar/core/map_reduce.hh

```diff
@@ -191,7 +191,8 @@ map_reduce(Iterator begin, Iterator end, Mapper&& mapper, Initial initial, Reduc
     while (begin != end) {
         ret = futurize_invoke(s->mapper, *begin++).then_wrapped([s = s.get(), ret = std::move(ret)] (auto f) mutable {
             try {
-                s->result = s->reduce(std::move(s->result), f.get());
+                auto mapped_value = f.get(); // Extract exception before touching s->result
+                s->result = s->reduce(std::move(s->result), std::move(mapped_value));
```

Comment:
Don't we have the same problem if reduce() throws?

Reply:
Yes, good catch